webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -975,6 +1098,83 @@ public static ByteArrayOutputStream requestsToZip(ServletContext context, String
 
         ZipOutputStream zipOut = new ZipOutputStream(baosZip);
         HashMap<String, Boolean> seenTitles = new HashMap<String, Boolean>();
+
+        boolean fmcBeingHeld = false;
+        for(ScrambleRequest scrambleRequest : scrambleRequests) {
+            if(scrambleRequest.fmc) {
+                fmcBeingHeld = true;
+
+                String safeTitle = toFileSafeString(scrambleRequest.title);
+                int salt = 0;
+                String tempNewSafeTitle = safeTitle;
+                while(seenTitles.get(tempNewSafeTitle) != null) {
+                    tempNewSafeTitle = safeTitle + " (" + (++salt) + ")";
+                }
+                safeTitle = tempNewSafeTitle;
+                seenTitles.put(safeTitle, true);
+
+                String pdfFileName = "pdf/" + safeTitle + " Scramble Cutout Sheet.pdf";
+                parameters.setFileNameInZip(pdfFileName);
+                zipOut.putNextEntry(null, parameters);
+
+                ByteArrayOutputStream pdfOut = new ByteArrayOutputStream();
+                Rectangle pageSize = PageSize.LETTER;
+                Document doc = new Document(pageSize, 0, 0, 75, 75);
+                PdfWriter docWriter = PdfWriter.getInstance(doc, pdfOut);
+
+                docWriter.setBoxSize("art", new Rectangle(36, 54, pageSize.getWidth()-36, pageSize.getHeight()-54));
+
+                doc.addCreationDate();
+                doc.addProducer();
+                if(globalTitle != null) {
+                    doc.addTitle(globalTitle);
+                }
+
+                doc.open();
+                addFmcScrambleCutoutSheet(docWriter, doc, scrambleRequest, globalTitle, 0);
+                doc.close();
+
+                // TODO - is there a better way to convert from a PdfWriter to a PdfReader?
+                PdfReader pdfReader = new PdfReader(pdfOut.toByteArray());
+                byte[] b = new byte[(int) pdfReader.getFileLength()];
+                pdfReader.getSafeFile().readFully(b);
+                zipOut.write(b);
+
+                zipOut.closeEntry();
+                break;
```

Comment:
This break is the reason why we're only generating *one* of the many necessary cutout sheets (see demo above).